resource_processor/vmss_porter/runner.py

```diff
@@ -213,7 +213,7 @@ async def get_porter_outputs(msg_body: dict, message_logger_adapter: logging.Log
     if returncode != 0:
         error_message = "Error context message = " + " ".join(err.split('\n'))
         message_logger_adapter.info(f"{get_installation_id(msg_body)}: Failed to get outputs with error = {error_message}")
-        return False, ""
+        return False, {}
```

Comment:
I guess this is fine, right?

![image](https://github.com/microsoft/AzureTRE/assets/1524451/eb474902-291b-4723-95b1-103710fc43ec)

This change as prompted more questions than it answers.

We appear to return `False`  in this situation, but on line 229 (where we get invalid JSON) we simply log this event and then return `True` on line 231.

Then, back on line 194, we appear to totally ignore the `True`/`False` flag, and we report that everything was successful.

This change will therefore further mask issues. I assume things are currently blowing up because the porter command failed, and your proposed change will hide this, and report success.

Perhaps I'm missing something that makes this an OK thing to do.

Reply:
Hmm, thanks for looking, correct. I was just concerned with the service bus message being valid.

I don't think the change makes things worse, because at the moment the operation just sticks in an updating state and the user isnt. However it does look like the True/False should be handled differently. 

Maybe needs a separate issue to ensure if it's false then the operation is marked as failed? Similar to:

https://github.com/microsoft/AzureTRE/blob/ad647cecb6b97220ade13a1195e2d81cebed4736/resource_processor/vmss_porter/runner.py#L179-L182

Reply:
Agreed. I think the pattern you've highlighted is exactly what's needed.

Couldn't the empty dictionary be replaced with something that held and error status message, and the `success_message` be changed to say something else if the return value is False? If we did that in this PR, that would unblock the issue you see *and* make things easier to debug when they fail, right? 

As you say, your change doesn't make things worse. If we really need to make this change now, then I'll approve, but it doesn't feel like it would be much more work to make it better.

Reply:
@martinpeck I've made some changes, haven't tested yet, but value feedback.

Reply:
Assuming testing is good, this looks good to me! Thanks!